pkg/cmd/cli/backuplocation/set.go

```diff
@@ -104,6 +107,11 @@ func (o *SetOptions) Run(c *cobra.Command, f client.Factory) error {
 		}
 	}
 
+	defalutBSLs, err := storage.GetDefaultBackupStorageLocations(context.Background(), kbClient, f.Namespace())
```

Comment:
Move this logic inside the `if` block in line 126 to avoid unnecessary BSL listing

Reply:
updated